<launch>

  	<!-- Static transforms -->
  <include file="$(find airsim_ros_pkgs)/launch/static_transforms_fast.launch"/>

  <node pkg="airsim_ros_pkgs" type="kinect_publisher_fast.py" name="kinect_publish_node"
    output="screen" >
  </node>

  <!-- <node pkg="airsim_ros_pkgs" type="camera_pose_publisher.py" name="camera_pose_node"
    output="screen" >
  </node> -->

  <!-- <node pkg="airsim_ros_pkgs" type="tf_listener.py" name="listener" /> -->

   <!-- <node pkg="airsim_ros_pkgs" type="camera_pose_publisher.py" name="camera_pose_publisher" output="screen"/> -->

  <node pkg="nodelet" type="nodelet" args="manager" name="depth_to_pointcloud_manager" output="screen"/>

  <node pkg="nodelet" type="nodelet" name="airsim_depth2cloud" args="load depth_image_proc/point_cloud_xyz depth_to_pointcloud_manager" output="screen">
		<remap from="image_rect" to="/camera/depth_registered/image_raw"/>
		<remap from="points" to="/airsim_node/drone_1/points"/>
		<remap from="camera_info" to="/camera/depth_registered/camera_info"/>
	</node>

  <node pkg="airsim_ros_pkgs" name="pointcloud_filter_node" type="pointcloud_filter" output="screen">
	<param name="target_frame_id" value="/map"/>
	<param name="cloud_topic" value="/airsim_node/drone_1/points"/>
	<param name="leaf_size" value="0.2"/>
	<param name="ground_high" value="1.58"/>
	<param name="Neighbors_numb" value="50"/>
	<param name="td_threshold" value="0.50"/>
  <param name="depth" value="50"/>
	</node>


  <!-- <node pkg="airsim_ros_pkgs" type="save_pcd" name="save_pcd"
    output="screen" >
  </node> -->
  
</launch>